### PR TITLE
Adds jazzy docs generation and gh-pages releases

### PR DESCRIPTION
**Pull request by flovilmart**

### CODE REVIEW

**Comment by montymxb** [comment]
This looks okay at first glance here. I'm a bit short on time but I'm leaving my review still pending so I'll get to it later tonight when I can read into the travis.yml in detail.

**Comment by flovilmart** [comment]
No problem! No rush!

File: .travis.yml [hunk 1]
```diff
@@ -1,46 +1,67 @@
...
+          set -o pipefail
+          xcodebuild build -workspace ParseLiveQuery.xcworkspace -sdk iphonesimulator -scheme ParseLiveQuery-iOS -configuration Debug -destination "platform=iOS Simulator,name=iPhone 4s" -destination "platform=iOS Simulator,name=iPhone 6 Plus" GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=YES GCC_GENERATE_TEST_COVERAGE_FILES=YES | xcpretty -c
+      after_success: bash <(curl -s https://codecov.io/bash)
+    - stage: test
+      env: OSX
+      script: 
+      - | 
+          set -o pipefail
+          xcodebuild build -workspace ParseLiveQuery.xcworkspace -sdk macosx -scheme ParseLiveQuery-OSX -configuration Debug GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=YES GCC_GENERATE_TEST_COVERAGE_FILES=YES | xcpretty -c
+      after_success: bash <(curl -s https://codecov.io/bash)
```
Comment (montymxb) [inline .travis.yml:28]: There's an identical `after_success` call above, but in a separate test stage for a separate environment. Is there a way to consolidate these two declarations of stage **test** with different scripts depending on the environment, but the same `after_success` result?

**Review by montymxb — APPROVED** [review]
Alright this looks good 👍 . The one nit is trivial and not really that important given in the overall scheme here. Something we can adjust later on (if we can that is).

**Comment by flovilmart** [comment]
I can have a look at consolidating the test steps later indeed

**Comment by montymxb** [comment]
My thoughts exactly 👍 .

**Pull request merged**